pkg/resource/bucket/hook.go

```diff
@@ -299,6 +363,60 @@ func (rm *resourceManager) syncAccelerate(
 	return nil
 }
 
+func (rm *resourceManager) newGetBucketACLPayload(
+	r *resource,
+) *svcsdk.GetBucketAclInput {
+	res := &svcsdk.GetBucketAclInput{}
+	res.SetBucket(*r.ko.Spec.Name)
+	return res
+}
+
+func (rm *resourceManager) newPutBucketACLPayload(
+	r *resource,
+) *svcsdk.PutBucketAclInput {
+	res := &svcsdk.PutBucketAclInput{}
+	res.SetBucket(*r.ko.Spec.Name)
+	if r.ko.Spec.ACL != nil {
+		res.SetACL(*r.ko.Spec.ACL)
+	}
+
+	if r.ko.Spec.GrantFullControl != nil {
+		res.SetGrantFullControl(*r.ko.Spec.GrantFullControl)
+	}
+	if r.ko.Spec.GrantRead != nil {
+		res.SetGrantRead(*r.ko.Spec.GrantRead)
+	}
+	if r.ko.Spec.GrantReadACP != nil {
+		res.SetGrantReadACP(*r.ko.Spec.GrantReadACP)
+	}
+	if r.ko.Spec.GrantWrite != nil {
+		res.SetGrantWrite(*r.ko.Spec.GrantWrite)
+	}
+	if r.ko.Spec.GrantWriteACP != nil {
+		res.SetGrantWriteACP(*r.ko.Spec.GrantWriteACP)
+	}
+
+	return res
+}
+
+func (rm *resourceManager) syncACL(
+	ctx context.Context,
+	r *resource,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.syncACL")
+	defer exit(err)
+	input := rm.newPutBucketACLPayload(r)
+
+	_, err = rm.sdkapi.PutBucketAclWithContext(ctx, input)
+	rm.metrics.RecordAPICall("UPDATED", "PutBucketAcl", err)
```

Comment:
s/UPDATED/UPDATE/

I noticed this typo throughout this file, btw... might clean those up while you're at it.